modules/diskq/tests/test_diskq_truncate.c

```diff
@@ -177,22 +177,21 @@ test_diskq_truncate_without_diskbuffer_used(void)
   });
 }
 
-
-int
-main(void)
+static void
+setup(void)
 {
   msg_init(TRUE); // internal messages will go to stderr(), msg_init() is idempotent
   app_startup();
 
   configuration = cfg_new_snippet();
-  assert_true(cfg_init(configuration), "cfg_init failed!");
-
-  // Diskbuffer is the part of disk-queue that is used only when qout is full
-  test_diskq_truncate_with_diskbuffer_used();
-  test_diskq_truncate_without_diskbuffer_used();
+  cr_assert(cfg_init(configuration), "cfg_init failed!");
```

Comment:
The `setup` and `teardown` phases are not designed to assert in them. `assert` and `expect` should be kept in the `Test` phase.

Reply:
okay. I will remove assert from setup and teardown.

Reply:
What about using other error handling, but keeping cfg_init in setup function?
Maybe C standard `assert` works in setup?
I don't think it's a good way to have init functions per test cases.

Reply:
@gaborznagy : ok, I will use standard assert.